feat: migration guide v2.0.0 to v3.0.0

huichiaotsou opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Description
<!-- Small description. What have you fixed or changed? -->


## Checklist
- [x] Ran Linting
- [x] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.
- [ ] Added an entry to the `CHANGELOG.md` file.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

RiccardoM commented on docs/cosmos-based/parser/migrations/v3.0.0.md:
````diff
@@ -0,0 +1,164 @@
+DROP TABLE transaction_old;
+DROP TABLE message_old;
+```
+
````
I think that here we should also tell the user to update the BDJuno service content to replace the `parse` command with `start` instead

RiccardoM reviewed (approved):
Everything looks good to me

huichiaotsou commented:
@RiccardoM I opened a PR for modifying the hasura metadata so that users can apply their custom actions handler URL: https://github.com/forbole/bdjuno/pull/379

Should we add this change? If yes I'll add a section in the doc to remind user of adding the URL as env variable while starting the docker container.

huichiaotsou commented:
> > @RiccardoM I opened a PR for modifying the hasura metadata so that users can apply their custom actions handler URL: [forbole/bdjuno#379](https://github.com/forbole/bdjuno/pull/379)
> > Should we add this change? If yes I'll add a section in the doc to remind user of adding the URL as env variable while starting the docker container.
> 
> Yes please do add it

Done, could you please review the file `docs/cosmos-based/parser/hasura.md` & `docs/cosmos-based/parser/migrations/v3.0.0.md#Hasura` 🙏🏼 @RiccardoM @MonikaCat

MonikaCat reviewed (dismissed):
LGTM

RiccardoM commented:
Please don't merge this yet as we are working on the last feature for v3.0.0. We will merge once that's ready

MonikaCat commented:
@huichiaotsou @RiccardoM I updated hasura and migration docs and added new instructions how to run hasura actions after we merge it with `bdjuno start` cmd, can you review it pls? ^^

huichiaotsou commented:
> @huichiaotsou @RiccardoM I updated hasura and migration docs and added new instructions how to run hasura actions after we merge it with `bdjuno start` cmd, can you review it pls? ^^

Just pushed a commit to exchange the place of `pricefeed` and `gRPC/RPC` and add actions node example in the config page.

RiccardoM commented:
Ok guys I'm 100% trusting you with this :joy: One thing I would like to have though is an official test: follow the updated migration guide on a BDJuno instance that needs to be updated , and make sure that by copy/pasting things everything works properly. Once that's working without any issue, we can publish the migration guide. I would like to avoid any possible problem that might come up by people following a guide that is not 100% complete

MonikaCat commented:
Cool thanks @ryuash! I will test it on dev server today as well

ryuash commented:
overall review of a fresh setup:
Option but i think it would help/ make it easier if we actually added the commands of how to create a database/ user/ and import schemas here

<img width="1027" alt="image" src="https://user-images.githubusercontent.com/42913823/163203922-0c346f0e-725c-4420-9a2d-7e53c2b837ad.png">

---

Who is adding all the `$` in the code blocks?? Is there a way to do this but exclude the `$` in the copy or..

---


I see the setup requires `make` this requires additional setup as most OS do not come with this default.

<img width="439" alt="image" src="https://user-images.githubusercontent.com/42913823/163204423-c6ee9e7c-7ce0-41f3-ba53-758dd8caf7fa.png">

---

In hasura actions can we have a curl test and the expected return structure so that we can see it works? I've gotten a few issues where the bottle neck was the hasura actions were setup correctly but ui suppressed all the errors. 

--- 

I think we need to specify that hasura actions will run on the same server as bdjuno (given that it's now merged together) in the event that the user decides to split up hasura and bdjuno in different servers? This part is kind of confusing now such that, hasura actions server need to be with bdjuno but the metadata + hasura itself can be located elsewhere if that makes sense

MonikaCat commented:
I think it depends, who are we targeting as our end users? Most of our users will have at least some experience setting up/managing servers and databases. However, if you think it's needed I can add more detailed guide including steps to set up server and database, so then anyone can run it. 
Also yeah I added $ to indicate shell command and coz some commands were missing it.
